tests/e2e/e2e_pod_client_server_test.go

```diff
@@ -132,3 +117,32 @@ var _ = DescribeTier1("Test HTTP traffic from 1 pod client -> 1 pod server", fun
 		})
 	})
 })
+
+func setupSource(sourceName string, withKubernetesService bool) *v1.Pod {
```

Comment:
I wonder if you only needed to add the `if` condition around the `CreateService` :)